src/patterns/ComposedModal/ComposedModal.tsx

```diff
@@ -0,0 +1,152 @@
+import classnames from 'classnames';
+import { forwardRef } from 'react';
+import Button from '../../components/Button/Button';
+import { ButtonVariants } from '../../components/Button/types';
+import { Divider } from '../../components/Divider';
+import Modal, { ModalProps } from '../../components/Modal/Modal';
+import { Text, TextVariants } from '../../components/Text';
+import { getCommonProps, noOp } from '../../utils';
+
+export interface ViewingDetailsButtonsProps {
+  /**
+   * Callback click function for button
+   */
+  onClick?: () => void | unknown;
+  /**
+   * Button label text
+   */
+  buttonLabel?: string;
+}
+
+export interface ComposedModalProps extends Omit<ModalProps, 'onClose' | 'role' | 'style'> {
+  /**
+   * Title for Composed Modal
+   */
+  title: string;
+  /**
+   * The content of the modal.
+   */
+  children: React.ReactNode;
+  /**
+   * onClose handler for the modal
+   */
+  onClose?: () => void;
+  /**
+   * Add any additional styles to the component
+   */
+  style?: React.CSSProperties;
+  /**
+   * Whether the modal should focus after render
+   */
+  shouldFocusAfterRender?: boolean;
+  /**
+   * Optional proprerty center align all text in the component
+   */
+  centerAlignText?: boolean;
+  /**
+   * Maximum height value for the modal body
+   */
+  maxHeightValue?: string;
+  /**
+   * Left Button Props
+   */
+  secondaryButton?: ViewingDetailsButtonsProps;
+  /**
+   * Right Button Props
+   */
+  primaryButton?: ViewingDetailsButtonsProps;
+  /**
+   * Disclaimer text for bottom of Viewings Details
+   */
+  disclaimer?: string;
```

Comment:
Can we make this and the title prop named `titleText` and `disclaimerText` to align with our convention in other components?